sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -932,6 +934,7 @@ private[hive] object HiveClientImpl {
       table.storage.serde.getOrElse("org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe"))
     table.storage.properties.foreach { case (k, v) => hiveTable.setSerdeParam(k, v) }
     table.properties.foreach { case (k, v) => hiveTable.setProperty(k, v) }
+    hiveTable.setProperty(CREATED_SPARK_VERSION, table.createVersion)
```

Comment:
seems better to do it in `HiveExternalCatalog`? this version string stuff is not related to hive.